### PR TITLE
Replaced calls to binary.Write with bb.WriteByte in Encode

### PR DESCRIPTION
**Pull request by peterhellberg**
**Additional changes:**
 
 - Added comment for Encode
 - Added Encode benchmark
 - Some minor formatting
 - Removed width, height, r1, g1, b1, a1 variables

### CODE REVIEW

**Comment by mehlon** [comment]
:+1:

File: encode_test.go [hunk 1]
```diff
@@ -28,3 +29,22 @@ func TestEncode(t *testing.T) {
 		t.Fatalf(`w.Bytes() != imagefileData`)
 	}
 }
+
+func BenchmarkEncode(benchmark *testing.B) {
+	m := image.NewRGBA(image.Rect(0, 0, 1000, 1000))
+
+	draw.Draw(m, m.Bounds(), &image.Uniform{g}, image.ZP, draw.Src)
```
Comment (opennota) [inline encode_test.go]: This benchmark probably needs `b.StopTimer()` and `b.StartTimer()`.

Comment (peterhellberg) [inline encode_test.go]: Yes, you are right and since you changed the name of the color variables then it should be changed to b.

Comment (opennota) [inline encode_test.go]: It was just written from scratch. I didn't notice the benchmark when I looked at the diff.

Comment (peterhellberg) [inline encode_test.go]: Ok, your changes did indeed remove almost all allocations.

**Comment by opennota** [comment]
> Also, wouldn’t a b.ResetTimer() before the for loop be enough?

Good point.

**Comment by peterhellberg** [comment]
I can open a benchmark PR if you want?

**Comment by opennota** [comment]
@peterhellberg Go ahead.

**Pull request merged**